Add support for Helm 3

Helm 3 was released in November 2019. We should make sure the plugin works with both Helm 2 and 3.

Thank you! It helped with download, but not with helm3 - next error is 
```
[INFO] Clear target directory to ensure clean target package
[INFO] Created target helm directory
[ERROR] Output: Error: unknown flag: --client-only
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -4,6 +4,7 @@
 
 * Support overriding the chart version using the configuration parameter `chartVersion`. If `chartVersion` is not
   configured, the value for `project.model.version` will be used instead.
+* Add support for Helm 3.
 
 == Version 2.2.0
 

--- a/README.adoc
+++ b/README.adoc
@@ -87,6 +87,7 @@ that the correct docker image is used. An example snippet:
 |helmGroupId |`"com.deviceinsight.helm"` |The helm binary `groupId`
 |helmArtifactId |`"helm"` |The helm binary `artifactId`
 |helmVersion |None |The helm binary `version`
+|helmDownloadUrl |`"https://get.helm.sh/"` |The URL where the helm binary is downloaded from.
 |helm.skip |`false` |If true, execution will be skipped entirely
 |strictLint |`false` |If true, linting fails on warnings (see: <<goal-lint>>)
 |valuesFile | None | values file that should be used for goals <<goal-lint>>, <<goal-template>>

--- a/pom.xml
+++ b/pom.xml
@@ -12,13 +12,13 @@
 	<url>https://github.com/deviceinsight/helm-maven-plugin</url>
 
 	<properties>
-		<kotlin.version>1.3.50</kotlin.version>
+		<kotlin.version>1.3.61</kotlin.version>
 		<kotlin.compiler.jvmTarget>1.8</kotlin.compiler.jvmTarget>
 		<maven.compiler.source>1.8</maven.compiler.source>
 		<maven.compiler.target>1.8</maven.compiler.target>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-		<maven.version>3.6.3</maven.version>
+		<maven.version>3.6.0</maven.version>
 		<maven-project.version>2.2.1</maven-project.version>
 		<maven-artifact-transfer.version>0.10.1</maven-artifact-transfer.version>
 

--- a/src/main/kotlin/com/deviceinsight/helm/AbstractHelmMojo.kt
+++ b/src/main/kotlin/com/deviceinsight/helm/AbstractHelmMojo.kt
@@ -48,7 +48,7 @@ abstract class AbstractHelmMojo : AbstractMojo() {
 	@Parameter(property = "helmVersion", required = true)
 	private lateinit var helmVersion: String
 
-	@Parameter(property = "helmDownloadUrl", defaultValue = "https://kubernetes-helm.storage.googleapis.com/")
+	@Parameter(property = "helmDownloadUrl", defaultValue = "https://get.helm.sh/")
 	private lateinit var helmDownloadUrl: URI
 
 	@Parameter(property = "chartVersion", required = false, defaultValue = "\${project.model.version}")

--- a/src/main/kotlin/com/deviceinsight/helm/AbstractHelmMojo.kt
+++ b/src/main/kotlin/com/deviceinsight/helm/AbstractHelmMojo.kt
@@ -119,6 +119,8 @@ abstract class AbstractHelmMojo : AbstractMojo() {
 		}
 	}
 
+	protected fun majorHelmVersion(): Int = helmVersion.splitToSequence('.').first().toInt()
+
 	protected fun target() = File(project.build.directory).resolve("helm")
 
 	protected fun chartTarGzFile() = target().resolve("${chartName()}-${chartVersion}.tgz")

--- a/src/main/kotlin/com/deviceinsight/helm/PackageMojo.kt
+++ b/src/main/kotlin/com/deviceinsight/helm/PackageMojo.kt
@@ -67,7 +67,9 @@ class PackageMojo : AbstractHelmMojo() {
 
 			processHelmConfigFiles(targetHelmDir)
 
-			executeCmd("$helm init --client-only")
+			if (majorHelmVersion() < 3) {
+				executeCmd("$helm init --client-only")
+			}
 			executeCmd("$helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com")
 			executeCmd("$helm repo add chartRepo $chartRepoUrl")
 			executeCmd("$helm dependency update", directory = targetHelmDir)

